match/match.go

```diff
@@ -73,6 +78,34 @@ func matchCreateHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	userOne := *req.UserOne
+	userTwo := *req.UserTwo
```

Comment:
I'm not sure about this, why don't we just dereference them when we need them?

Reply:
@smithwjv What's your thoughts?

Reply:
I'm easy on this, motivation was to avoid dereferencing them twice each, but it doesn't save code

Reply:
Yeah I think we should dereference at use, as we pass the `req` object to `dao.CreateMatch`.